af/fastpath/fastpath/core.py

```diff
@@ -285,96 +303,114 @@ def process_measurements_from_s3() -> None:
                 return
 
 
+def minifp(fp: Fingerprint) -> Dict[str, Any]:
+    fields = (
+        "name",
+        "scope",
+        "location_found",
+        "confidence_no_fp",
+        "expected_countries",
+    )
+    return {k: v for k, v in fp.items() if k in fields}
+
+
+def match_http_body_fingerprints(resp, matches) -> None:
+    # Match HTTP body if found
+    body = resp.get("body")
+    if isinstance(body, dict):
+        if "data" in body and body.get("format", "") == "base64":
+            log.debug("Decoding base64 body")
+            body = b64decode(body["data"])
+            # returns bytes. bm.encode() below is faster that decoding it
+        else:
+            logbug(2, "incorrect body of type dict", {})
+            return
+
+    if body is None:
+        return
+
+    for fp in fingerprints["http"]:
+        if fp["location_found"] != "body":
+            continue
+        tb = time.time()
+        bm = fp["pattern"]
+        if isinstance(body, bytes):
+            idx = body.find(bm.encode())
```

Comment:
This will lead us to missing some fingerprints in the event of response bodies in some non-utf8 encoding. I have seen this happen before and is the reason why I have to do this quite sketching thing in ooni/data: https://github.com/ooni/data/blob/main/oonidata/datautils.py#L74.

Doing this though means we have to pay a pretty high tax for decoding all bodies, so probably we should not do this in the fastpath, but I thought it would still be useful to mention in this context as something to keep in mind.